### PR TITLE
Api continent filter

### PR DESCRIPTION
**Pull request by fm1psy**
Added an option to filter earthquake data from one of seven continents:
North America
South America
Asia
Africa
Oceania
Europe
Antarctica

### CODE REVIEW

File: api/api.py [hunk 1]
```diff
@@ -58,10 +59,41 @@ def get_filter_queries(earthquake_filters: dict[str]) -> list[str]:
...
+            country_code = location["country_code"]
+            # get continent code from country code
+            continent_code = pc.country_alpha2_to_continent_code(country_code)
+            if continent_code == continent_filter_code:
+                res.append(row)
+                print("row added!")
+            print("run complete!")
+        except Exception as e:  # pylint: disable=broad-exception-caught
+            print(str(e))
     return res
```
Comment (joe1606) [inline api/api.py:97]: Just wondering it its possible to break this function down? not entirely sure if its possible but its quite a large function which does several things

File: api/api.py [hunk 2]
```diff
@@ -58,10 +59,41 @@ def get_filter_queries(earthquake_filters: dict[str]) -> list[str]:
...
+        try:
+            location = rg.get((row["lat"], row["lon"]))
+            # extract country code
+            country_code = location["country_code"]
+            # get continent code from country code
+            continent_code = pc.country_alpha2_to_continent_code(country_code)
+            if continent_code == continent_filter_code:
+                res.append(row)
+                print("row added!")
+            print("run complete!")
```
Comment (joe1606) [inline api/api.py:94]: could use logging instead of prints statements maybe

File: api/test_api.py [hunk 3]
```diff
@@ -134,6 +133,11 @@ def test_get_filter_queries_alert_is_first_filter():
...
-        "e.magnitude >= 'min mag'",
-        "NOT YET IMPLEMENTED"
+        "e.magnitude >= 'min mag'"
     ]
+
+
+def test_filter_by_continent(example_api_response):
+    assert filter_by_continent(example_api_response, "Africa") == []
+    assert filter_by_continent(
+        example_api_response, "North America") == example_api_response
```
Comment (joe1606) [inline api/test_api.py:143]: you can try using pytest.parametarize to test several inputs and outputs in a single function

**Review by joe1606 — COMMENTED** [review]
good work, just a few comments on code quality. Also, may be worth thinking about moving your fixtures into a conftest file

File: api/api.py (hunk 1 above)
Comment (fm1psy) [reply api/api.py:97]: I was thinking the same. I may have the chance to do this once I do the country filter

File: api/api.py (hunk 2 above)
Comment (fm1psy) [reply api/api.py:94]: This was done for debugging, and should not be part of the final implementation. Thanks for pointing this out.

File: api/test_api.py (hunk 3 above)
Comment (fm1psy) [reply api/test_api.py:143]: True, I will consider this when I finalise testing.

File: api/api.py (hunk 1 above)
Comment (joe1606) [reply api/api.py]: sounds good to me

**Review by joe1606 — APPROVED** [review]
looking good, understandable decisions on the code you have so far.

**Pull request merged**